lib/mistral_ai_summary_example/settings_dialogs.dart

```diff
@@ -162,16 +128,50 @@ class _RandomSeedDialogState extends State<RandomSeedDialog> {
 
   @override
   Widget build(BuildContext context) {
-    return BaseSettingsDialog(
+    return BaseNumberInputDialog(
       title: 'Random Seed',
       description: 'Write a number if you want to set a random seed',
+      hintText: 'e.g. 123',
+      controller: _randomSeedController,
       onSettingChanged: () {
         final randomSeed = _randomSeedController.text.isEmpty
             ? null
             : int.tryParse(_randomSeedController.text);
         context.read<SummarySettingsModel>().setRandomSeed(randomSeed);
         Navigator.of(context).pop();
       },
+      onInputChanged: (value) => setState(() {}),
+      onClear: () => setState(_randomSeedController.clear),
+    );
+  }
+}
+
+class BaseNumberInputDialog extends StatelessWidget {
+  const BaseNumberInputDialog({
+    required this.title,
+    required this.description,
+    required this.hintText,
+    required this.controller,
+    required this.onSettingChanged,
+    required this.onInputChanged,
+    required this.onClear,
+    super.key,
+  });
+
+  final String title;
+  final String description;
+  final String hintText;
+  final TextEditingController controller;
+  final VoidCallback onSettingChanged;
```

Comment:
If this is a number input dialog then maybe it would be good to pass an `int` into `onSettingChanged`. You will have to move the int parsing into the  `BaseNumberInputDialog` but will remove duplication in specific setting widgets that use it.

Reply:
Good point refactored in [3d385e3](https://github.com/nomtek/flutter_ai_examples/pull/30/commits/3d385e337db8e0df7f9b280348d460d2412259fb)